### ISSUE TITLE
[Phase 3 | Feature 1] Prakriti AI Analyzer – Tongue, Nail & Face Image-Based Self-Assessment

### ISSUE THREAD

**Issue by Venky2834kami** [Venky2834kami/AyurTime#15]
## 🧬 Overview
Build a multi-modal AI-powered Prakriti (body constitution) analyzer that uses images of the user's tongue, nails, and face to identify their Ayurvedic dosha type (Vata, Pitta, Kapha) and generate a personalized health profile.

---

## 🎯 Goals
- Allow users to take/upload photos of tongue, nail, and face
- Run inference via a pre-trained or fine-tuned vision model
- Output dosha classification with confidence scores
- Display personalized Ayurvedic recommendations based on Prakriti type

---

### Frontend (UI)
- [ ] Camera capture UI for tongue, nail, face (3 steps)
- [ ] Image upload fallback with preview
- [ ] Step-by-step guided assessment wizard
- [ ] Results screen: Dosha pie chart + trait summary
- [ ] Save Prakriti profile to user account

### AI Model Integration
- [ ] Define image classification schema: input = image, output = `{vata_pct, pitta_pct, kapha_pct}`
- [ ] API endpoint: `POST /api/prakriti/analyze` — accepts multipart images, returns dosha scores
- [ ] Model: Use lightweight MobileNetV2 or EfficientNet-B0 fine-tuned on Ayurvedic tongue/face datasets
- [ ] Fallback: Rule-based questionnaire if camera is unavailable

### Data Schema (Prakriti Profile)
```json
{
  "user_id": "string",
  "analyzed_at": "ISO8601",
  "images": {
    "tongue": "url",
    "nail": "url",
    "face": "url"
  },
  "dosha_scores": {
    "vata": 0.45,
    "pitta": 0.35,
    "kapha": 0.20
  },
  "dominant_dosha": "vata",
  "sub_dosha": "vata-pitta",
  "prakriti_notes": "string"
}
```

---

## ✅ Acceptance Criteria
- [ ] User can upload 3 images (tongue, nail, face) and receive a dosha report within 10 seconds
- [ ] Results display dominant dosha with a breakdown chart
- [ ] Prakriti profile is saved and viewable from the user's dashboard
- [ ] Questionnaire fallback works without image upload
- [ ] Mobile-responsive UI with step-by-step guidance in English + Hindi
- [ ] API returns error gracefully if image quality is too low

---

## 🔗 Dependencies
- Pre-trained vision model or dataset (Kaggle / custom labeling)
- User authentication module (Phase 1/2)
- Dosha recommendation engine (text or LLM-based)

---

## 🏷️ Labels
`feature` `phase-3` `AI/ML` `prakriti` `priority:high`

---

## 📅 Target Sprint
July Week 1–3, 2026
